### PR TITLE
Story/sus 15

### PR DESCRIPTION
**Pull request by namrathaov**
Changed URI in search page to display title and year by loading dynamically.

### CODE REVIEW

File: sustainability/src/main/java/edu/asu/diging/sustainability/web/search/ResearcherSearchController.java [hunk 1]
```diff
@@ -37,7 +60,19 @@ public String search(Model model, @RequestParam("selectedConcepts") String[] sel
...
     }
+
+    @RequestMapping(value = "/perspective/researcher/resource", method = RequestMethod.GET)
+    @ResponseBody
+    public IResource reload(@RequestParam("uri") String uri) {
+        IResource resource = null;
+        try {
+            resource = resourceManager.getResource(uri);
+        } catch (NotAValidResourceException e) {
+            logger.error("Resource " + uri + " is not a valid one.");
```
Comment (jdamerow) [inline sustainability/src/main/java/edu/asu/diging/sustainability/web/search/ResearcherSearchController.java:77]: don't swallow the exception!

File: sustainability/src/main/java/edu/asu/diging/sustainability/web/search/ResearcherSearchController.java [hunk 2]
```diff
@@ -2,33 +2,56 @@
...
 
-    @RequestMapping(value="/perspective/search", method=RequestMethod.POST)
+    @Autowired
+    private IResourceManager resourceManager;
+
+    @RequestMapping(value = "/perspective/search", method = RequestMethod.POST)
     public String search(Model model, @RequestParam("selectedConcepts") String[] selectedConcepts) {
-        model.addAttribute("results", annotationManager.findTextsForConcepts(selectedConcepts));
-        
+        Map<String, List<IAnnotation>> URIList =
```
Comment (jdamerow) [inline sustainability/src/main/java/edu/asu/diging/sustainability/web/search/ResearcherSearchController.java]: variable names **never** start with a capital letter except when they are constants!

File: sustainability/src/main/java/edu/asu/diging/sustainability/web/search/ResearcherSearchController.java [hunk 3]
```diff
@@ -2,33 +2,56 @@
...
+                annotationManager.findTextsForConcepts(selectedConcepts);
+        List<IResource> resourceList = new ArrayList<IResource>();
+        for (String uri : URIList.keySet()) {
+            try {
+                resourceList.add(resourceManager.getResource(uri));
+            } catch (NotAValidResourceException e) {
+                logger.error("Resource " + uri + " is not a valid one.");
```
Comment (jdamerow) [inline sustainability/src/main/java/edu/asu/diging/sustainability/web/search/ResearcherSearchController.java:55]: don't swallow the exception!

File: sustainability/src/main/webapp/WEB-INF/tiles-defs.xml [hunk 4]
```diff
@@ -63,7 +63,9 @@
 	<definition name="perspective/researcher/search" extends="base.definition">
 		<put-attribute name="content" value="/WEB-INF/views/search/results.jsp" />	
 	</definition>
-	
+	<definition name="perspective/researcher/resource" extends="base.definition">
```
Comment (jdamerow) [inline sustainability/src/main/webapp/WEB-INF/tiles-defs.xml]: why did you add that here. this endpoint doesn't return a jsp.

File: sustainability/src/main/webapp/WEB-INF/views/search/results.jsp [hunk 5]
```diff
@@ -13,9 +13,48 @@
...
+				{
+				    $.ajax({
+					  url: "<c:url value="/perspective/researcher/resource"/>",
+					  type : 'GET',
+					  data: { 'uri' : uri },
+					  success: function (rsrc) {
+					    setTimeout(function(){
+					      if(rsrc.updatedOn != null){
+					        var updateResults = "<li class = 'list-group-item'>" +
+					          "<a href = '<c:url value = '/text?uri = " + rsrc.uri + "'/>'>"
```
Comment (jdamerow) [inline sustainability/src/main/webapp/WEB-INF/views/search/results.jsp]: this cannot work. did you test that? c:url is a jsp expression evaluated server-side. at this point you do some javascript which is client-side.

File: sustainability/src/main/webapp/WEB-INF/views/search/results.jsp [hunk 6]
```diff
@@ -13,9 +13,48 @@
...
+			                      + rsrc.title + " - " +	rsrc.year + "</a></li>";
```
Comment (jdamerow) [inline sustainability/src/main/webapp/WEB-INF/views/search/results.jsp]: years should be in parentheses like everywhere else.

File: sustainability/src/main/webapp/WEB-INF/views/search/results.jsp [hunk 7]
```diff
@@ -13,9 +13,48 @@
...
+			</li>
+			<script>
+				setTimeout(reload('${entry.uri}','${vs.index}'), 3000);
+				function reload(uri, index)
+				{
+				    $.ajax({
+					  url: "<c:url value="/perspective/researcher/resource"/>",
+					  type : 'GET',
+					  data: { 'uri' : uri },
+					  success: function (rsrc) {
```
Comment (jdamerow) [inline sustainability/src/main/webapp/WEB-INF/views/search/results.jsp:30]: what happens if the controller returns null?

File: sustainability/src/main/webapp/WEB-INF/views/search/results.jsp [hunk 8]
```diff
@@ -13,9 +13,48 @@
...
+						  }
+					    },3000);
+				      },
+				      error: function(e){
+				    	  var updateResults = "<li class = 'list-group-item'>" +
+				    	    "<i class='glyphicon glyphicon-remove-sign'></i>" + 
+				    	      " Error loading resource for ${entry.uri}</li"
+				    	  $('#part' + index).replaceWith(updateResults);
+					  }
+				    });
```
Comment (jdamerow) [inline sustainability/src/main/webapp/WEB-INF/views/search/results.jsp]: formatting is all over the place

**Review by jdamerow — CHANGES_REQUESTED** [review]

**Comment by namrathaov** [comment]
[SUS-15] Minor fixes, exception handling and formatting.

File: sustainability/src/main/java/edu/asu/diging/sustainability/web/search/ResearcherSearchController.java (hunk 3 above)
Comment (jdamerow) [reply sustainability/src/main/java/edu/asu/diging/sustainability/web/search/ResearcherSearchController.java:55]: not addressed

File: sustainability/src/main/java/edu/asu/diging/sustainability/web/search/ResearcherSearchController.java [hunk 9]
```diff
@@ -37,7 +64,19 @@ public String search(Model model, @RequestParam("selectedConcepts") String[] sel
             }
         }
         model.addAttribute("concepts", concepts);
-        
         return "perspective/researcher/search";
     }
+
+    @RequestMapping(value = "/perspective/researcher/resource", method = RequestMethod.GET)
+    @ResponseBody
```
Comment (jdamerow) [inline sustainability/src/main/java/edu/asu/diging/sustainability/web/search/ResearcherSearchController.java:71]: remove. shouldn't be necessary anymore if you return a response entity

File: sustainability/src/main/java/edu/asu/diging/sustainability/web/search/ResearcherSearchController.java (hunk 1 above)
Comment (jdamerow) [reply sustainability/src/main/java/edu/asu/diging/sustainability/web/search/ResearcherSearchController.java:77]: not addressed

File: sustainability/src/main/webapp/WEB-INF/views/search/results.jsp [hunk 10]
```diff
@@ -13,9 +13,51 @@
...
+					  	setTimeout(function() {
+							if(rsrc.updatedOn != null) {
+							  var updateResults = "<li class = 'list-group-item'>" +
+								"<c:url value='/text?uri=' var='rsrcUrl' />" +
+									"<a href = ${rsrcUrl}" + rsrc.uri + ">" + 
+										rsrc.title + " (" +	rsrc.year + 
+											")</a></li>";
+							$('#part' + index).replaceWith(updateResults);
+						  }
+						  else if(rsrc.updatedOn == null){
```
Comment (jdamerow) [inline sustainability/src/main/webapp/WEB-INF/views/search/results.jsp:40]: why do you have an extra if here? is there another option besides == null or != null?

**Review by jdamerow — CHANGES_REQUESTED** [review]

**Pull request closed without merging**